docs/configure.md

```diff
@@ -377,7 +377,70 @@ See [the official documentation](https://kubernetes.io/docs/concepts/configurati
+**Note:** The storage class configuration steps should be done _before_ running `./kubectl-apply-all.sh` for the first time. Otherwise, you'll have to manually migrate the PVCs that are already bound to the default storage class to the custom one that you specified.
+
+By default, Sourcegraph uses the default storage class of your cluster. However, **we highly recommend that you use a storage class that uses SSDs as the underlying disk type**. The configuration details for the storage class differs depending on your hosting provider, so you should:
+
+1. Create a stub `base/default.StorageClass.yaml`.
+
+   ```yaml
+   # base/default.StorageClass.yaml
+   kind: StorageClass
+   apiVersion: storage.k8s.io/v1
+   metadata:
+     name: default
+     labels:
+       deploy: sourcegraph
+     annotations:
+       storageclass.kubernetes.io/is-default-class: "true"
+   #
+   # The values of the "provisioner" and "parameters" fields will differ depending on the cloud provider that you are using. Please read through https://kubernetes.io/docs/concepts/storage/storage-classes/ in order to know what values to add. We recommend specifying SSDs as the disk type if possible.
+   #
+   # For example, if you are using GKE with a cluster whose nodes are all in the "us-central1-a" zone, you could use the following values:
+   #
+   # provisioner: kubernetes.io/gce-pd
+   # parameters:
+   #  type: pd-ssd
+   #  zones: us-central1-a
+   ```
+
+1. Read through the [Kubernetes storage class documentation](https://kubernetes.io/docs/concepts/storage/storage-classes/), and fill in the `provisioner` and `parameters` fields in `base/default.StorageClass.yaml` with the correct values for your hosting provider (e.x.: [GCP](https://kubernetes.io/docs/concepts/storage/storage-classes/#gce), [AWS](https://kubernetes.io/docs/concepts/storage/storage-classes/#aws), [Azure](https://kubernetes.io/docs/concepts/storage/storage-classes/#azure-disk)). **Again, we highly recommend that the storage class use SSDs as the underlying disk type.**
+
+1. [Mark the existing storage classes in your cluster as non-default](https://kubernetes.io/docs/tasks/administer-cluster/change-default-storage-class/#changing-the-default-storageclass):
+
+   ```bash
+    kubectl get sc -o json | jq --raw-output ".items | map(select(.metadata.name)) | .[] | \"kubectl patch sc -p '{\\\"metadata\\\":{\\\"annotations\\\":{\\\"storageclass.kubernetes.io/is-default-class\\\":\\\"false\\\"}}}' \\(.metadata.name)\"" | bash
+   ```
+
+1. In [create-new-cluster.sh](../create-new-cluster.sh), add the above command _before_ `./kubectl-apply-all`.
+
+   ```bash
+   #!/bin/bash
+
+   # Mark the existing storage classes in the cluster as non-default
+   kubectl get sc -o json | jq --raw-output ".items | map(select(.metadata.name)) | .[] | \"kubectl patch sc -p '{\\\"metadata\\\":{\\\"annotations\\\":{\\\"storageclass.kubernetes.io/is-default-class\\\":\\\"false\\\"}}}' \\(.metadata.name)\"" | bash
```

Comment:
Actually, this won't work if you have a storage class with the `"storageclass.beta.kubernetes.io/is-default-class` annotation. I'm trying to figure out a jq script that can handle both cases.